### PR TITLE
Feat/ecs/show hit box

### PR DESCRIPTION
**Pull request by Bootoyka**
### Hitbox Feature

- [x] Gives the player the possibility to change display the Hitbox by pressing H by default
- [x] The keyBind is mappable like others in controllable
- [x] The first world is ready to move out the entryPoint

### CODE REVIEW

**Review by GuyomT — APPROVED** [review]
NTR

**Review by Cavonstavant — APPROVED** [review]
seems ok

File: src/ecs/World.hpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace ecs
          * Used to get the first event of the private stack _events
          * @return The first event of the private stack if it exists, ecs::Event::EventType::Null otherwise
          */
-        const ecs::Event getEvent() const
+        ecs::Event getEvent() const
```
Comment (aurelenc) [inline src/ecs/World.hpp:52]: const

File: src/ecs/World.hpp [hunk 2]
```diff
@@ -60,7 +60,7 @@ namespace ecs
          * Used to pop the first event of the private stack _events
          * @return The event you just popped out of the stack if it exists, ecs::Event::EventType::Null otherwise
          */
-        const ecs::Event popEvent()
+        ecs::Event popEvent()
```
Comment (aurelenc) [inline src/ecs/World.hpp:63]: const

File: src/ecs/systems/client/HandleIncomingMessages.hpp [hunk 3]
```diff
@@ -76,26 +76,24 @@ namespace ecs::systems
...
+                        newEntity, {sf::Keyboard::Z, sf::Keyboard::Q, sf::Keyboard::S, sf::Keyboard::D, sf::Keyboard::H});
+                    world.registry.addComponent<ecs::component::Hitbox>(
+                        newEntity, {ecs::component::Hitbox()});
                 }
                 world.registry.addComponent<component::Drawable>(newEntity, {"players", {1, 1, 32, 16}});
                 break;
             case component::EntityType::Types::EnemyBase:
                 world.registry.addComponent<component::Drawable>(newEntity, {"players", {1, 18, 32, 16}});
+                world.registry.addComponent<ecs::component::Hitbox>(
+                    newEntity, {ecs::component::Hitbox()});
```
Comment (aurelenc) [inline src/ecs/systems/client/HandleIncomingMessages.hpp:88]: you could just have passed {} instead of {ecs::component::Hitbox()}

**Review by Cavonstavant — APPROVED** [review]

File: src/ecs/systems/client/HandleIncomingMessages.hpp (hunk 3 above)
Comment (Bootoyka) [reply src/ecs/systems/client/HandleIncomingMessages.hpp:88]: true

Comment (Bootoyka) [reply src/ecs/systems/client/HandleIncomingMessages.hpp:88]: Nop

**Pull request merged**